ThoR/Alloy/Syntax/namespaceSeparator.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2024 RheinMain University of Applied Sciences
+Released under license as described in the file LICENSE.
+Authors: s. file CONTRIBUTORS
+-/
+import ThoR.Basic
+open Lean
+
+/-
+  Represents an alloy-like namespace
+-/
+namespace Alloy
+
+  /--
+  This Structure represents the ident value of the alloy-like namespace
+  -/
+  structure namespaceSeparator where
+    (representedNamespace : Ident)
+  deriving Repr
+
+  /--
+  A syntax repreaentation of an namespaceSeparator
+  -/
+  declare_syntax_cat namespaceSeparator
+  declare_syntax_cat namespaceSeparatorExtension
+  syntax "/" ident : namespaceSeparatorExtension
+  syntax ident (namespaceSeparatorExtension)*: namespaceSeparator
```

Comment:
How difficult is it to add protected Lean keywords (e.g. "variable", "def"; and all command names in general) to the syntax category and the subsequent matches?

Is it e.g. possible to provide a Lean array `#["variable", "def", ...]` which is then transformed into an addition to the syntax category? How can the elements of the list be handled in a unified way in the subsequent matches?

Reply:
I could not find a way to create Syntax from a list. However, I made it so you only have to define the keywords and their meaning once (see extendedIdent).